### DISCUSSION TITLE
QSPI Memory mapped flash on STM32F7508_DK

### DISCUSSION THREAD

**Discussion by Dermiste** [zephyrproject-rtos/zephyr#65861] [Q&A] [answered]
Hi all,

I am using zephyr on a **STM32F7508_DK** board which only has **64k flash**, such memory space won't hold most networking example but the **16MB of QSPI nor** would hold much more ! The process of using that one instead of the flash as ROM for my program would be "memory mapping" but I cannot find any help on that, which means it's either super trivial (maybe just using the device tree?) or super complicated.

Does anyone know how to go about with such a thing? any help/link/tip much appreciated.

Bests

**Comment by erwango** [comment]
That's an on going effort. See https://github.com/zephyrproject-rtos/zephyr/pull/64641

**Comment by FRASTM** [comment]
To complete.  The stm32 qspi in memory mapped mode on the stm32f7 serie has not been tested (unfortunately running on the stm32H7 revealed some issues, see https://github.com/zephyrproject-rtos/zephyr/pull/61122 )
You have to take the update stm32 qspi driver which sets the MemoryMapped mode at the end of the initialization.
You have to build and link your application to be executed in external NOR FLASH. 
Then flash it in the external NOR flash.
Then execute it : this is what the ../bootloader/mcuboot/boot/zephyr/ does in the do_boot()  when the memory partitioning is defining the `slot1_partition` in the external NOR flash.

**Comment by Dermiste** [comment]
Thanks for the links.

Does it mean that at some point if the PR gets accepted it will be possible to set this up only using device tree tweaks?

**Comment by FRASTM** [comment]
I have a branch https://github.com/FRASTM/zephyr/tree/qspi_init where the drivers/flash/flash_stm32_qspi.c is enabling the MEMMAP mode.

**Comment by davidschneider-cpi** [comment, marked as answer]
This feature is now merged and will be part of Zephyr 3.7 release: https://github.com/zephyrproject-rtos/zephyr/pull/72339
